### PR TITLE
chore: introduce apps/web-tools ported from common-tools

### PR DESCRIPTION
**Pull request by gre**
### 📝 Description

- [x] Re-create the web project known as "common-tools" as a Next.js fresh project in `apps/web-tools`.
- [x] Port all the JavaScript into TypeScript (including some type fixes).

Future step:
- once this PR is merged, we will do a second PR that completely remove out "common-tools", at the moment it wasn't done yet because we still need Vercel deployment to happen, but we will then switch to apps/web-tools.

### ❓ Context

- **Impacted projects**: `none` 
- **Linked resource(s)**: LIVE-9317 

### ✅ Checklist

- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### CODE REVIEW

File: libs/ledger-live-common/package.json [hunk 1]
```diff
@@ -281,7 +277,7 @@
     "@types/lodash": "^4.14.179",
     "@types/node": "^18",
     "@types/object-hash": "^2.1.0",
-    "@types/react": "^17.0.53",
+    "@types/react": "17",
```
Comment (gre) [inline libs/ledger-live-common/package.json:280]: live-common actually seems incompatible with `@types/react@18` around some specific issue around the Provider typing. so meanwhile we will force that version (not having it as a peer dep) but we will come back to this issue when we need to port to v18 everywhere.

File: apps/web-tools/pages/_app.tsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import "./globals.css";
+import type { AppProps } from "next/app";
+
+export default function MyApp({ Component, pageProps }: AppProps) {
+  // @ts-expect-error FIXME something wrong with typing here
```
Comment (gre) [inline apps/web-tools/pages/_app.tsx:5]: I didn't figure this out but this code, suggested by the doc, have some type issue. I suspect it is due to React 17 vs 18 typing coercing issue. but i couldn't really prove it.

File: apps/web-tools/live-common-setup.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { setSupportedCurrencies } from "@ledgerhq/live-common/lib/currencies/index";
+import { setWalletAPIVersion } from "@ledgerhq/live-common/lib/wallet-api/version";
+import { WALLET_API_VERSION } from "@ledgerhq/live-common/lib/wallet-api/constants";
+
+setWalletAPIVersion(WALLET_API_VERSION);
+
+setSupportedCurrencies([
```
Comment (haammar-ledger) [inline apps/web-tools/live-common-setup.ts:7]: A few chains may be missing here compared to LLD and LLM on develop

File: apps/web-tools/package.json [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    "@ledgerhq/coin-framework": "workspace:^",
+    "@ledgerhq/domain-service": "workspace:^",
+    "@ledgerhq/errors": "workspace:^",
+    "@ledgerhq/hw-app-eth": "workspace:^",
+    "@ledgerhq/hw-transport": "workspace:^",
+    "@ledgerhq/hw-transport-http": "workspace:^",
+    "@ledgerhq/hw-transport-web-ble": "workspace:^",
+    "@ledgerhq/hw-transport-webhid": "workspace:^",
+    "@ledgerhq/hw-transport-webusb": "workspace:^",
+    "@ledgerhq/live-common": "workspace:^",
```
Comment (haammar-ledger) [inline apps/web-tools/package.json:27]: Aren't `coin-*` libs required too? To support algorand + polkadot + evm that have already been extracted?

File: apps/web-tools/pages/sync.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
+"use client";
```
Comment (haammar-ledger) [inline apps/web-tools/pages/sync.tsx:1]: Curious, what is this for?

**Review by haammar-ledger — COMMENTED** [review]
This is awesome :rocket: 

Just a suggestion, both for easier UX and DX : a short README for each of the tool, briefly explaining what it is for + how it works.

File: apps/web-tools/live-common-setup.ts (hunk 3 above)
Comment (gre) [reply apps/web-tools/live-common-setup.ts:7]: ( and i just took the same from common-tools )

File: apps/web-tools/package.json (hunk 4 above)
Comment (gre) [reply apps/web-tools/package.json:27]: ( same as in common-tools ) nope live-common is still a monolith that aggregate the coin together (so what you said is done at the live-common level)

File: apps/web-tools/pages/sync.tsx (hunk 5 above)
Comment (gre) [reply apps/web-tools/pages/sync.tsx:1]: https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs

I'm not even sure if it is still necessary, but at some point, without it I was getting compiling issues.

File: apps/web-tools/package.json (hunk 4 above)
Comment (haammar-ledger) [reply apps/web-tools/package.json:27]: Ah true, no direct dependency

File: apps/web-tools/live-common-setup.ts (hunk 3 above)
Comment (haammar-ledger) [reply apps/web-tools/live-common-setup.ts:7]: Which was probably outdated... it's on us so my bad, but this is an opportunity to update the list I think.
Another proof we'll want to unify this list of supported coins, coming soon with coin config :+1:

**Review by haammar-ledger — APPROVED** [review]

File: apps/web-tools/live-common-setup.ts (hunk 3 above)
Comment (haammar-ledger) [reply apps/web-tools/live-common-setup.ts:7]: Not blocking anyway, can be done later

**Comment by haammar-ledger** [comment]
Ad discussed, documenting each tool will be done in a second step soon

**Pull request merged**